lib/src/features/main/theme/theme_data/app_theme_data.dart

```diff
@@ -0,0 +1,283 @@
+import 'package:flutter/material.dart';
+import 'package:gui/gen/fonts.gen.dart';
+import 'package:gui/src/core/enums/theme_modes.dart';
+import 'package:gui/src/features/main/theme/theme_data/pallets/on_surface_pallet.dart';
+import 'package:gui/src/features/main/theme/theme_data/pallets/surface_pallet.dart';
+
+/// ## [AppThemeData] Documentation
+///
+/// The [AppThemeData] class provides theming configurations for the
+/// application.
+/// It contains a static map that holds the `ThemeData` for both light and dark
+/// modes, allowing for easy theme management based on the user's preference.
+///
+class AppThemeData {
+  const AppThemeData._();
+  static final Map<ThemeModes, ThemeData> themeDataModes = {
+    // light ThemeData.
+    ThemeModes.light: ThemeData(
+      bottomNavigationBarTheme: BottomNavigationBarThemeData(
+        backgroundColor: SurfacePallet.light.surface3,
+      ),
+      scaffoldBackgroundColor: SurfacePallet.light.surface3,
+      brightness: Brightness.light,
+      dividerColor: OnSurfacePallet.light.onSurface1,
+      textTheme: lightTextTheme,
+      cardColor: SurfacePallet.light.surface3,
+      extensions: const <ThemeExtension<dynamic>>[
+        OnSurfacePallet.light,
+        SurfacePallet.light,
+      ],
+    ),
+
+    // dark ThemeData.
+    ThemeModes.dark: ThemeData(
+      brightness: Brightness.dark,
+      textTheme: darkTextTheme,
+      extensions: <ThemeExtension<dynamic>>[
+        OnSurfacePallet.dark,
+        SurfacePallet.dark,
+      ],
+    ),
+  };
+
+  static final lightTextTheme = TextTheme(
+    displaySmall: TextStyle(
+      fontFamily: FontFamily.inter,
+      color: SurfacePallet.light.surface3,
+      fontStyle: FontStyle.normal,
+      fontWeight: FontWeight.w700,
+      fontSize: 16,
+      height: 24 / 12,
+      leadingDistribution: TextLeadingDistribution.even,
+    ),
+    headlineLarge: TextStyle(
+      fontFamily: FontFamily.inter,
+      color: SurfacePallet.light.surface3,
+      fontStyle: FontStyle.normal,
+      fontWeight: FontWeight.w300,
+      fontSize: 36,
+      height: 52 / 36,
+      leadingDistribution: TextLeadingDistribution.even,
+    ),
```

Comment:
Please create a method to reduce code duplication. For ex
 
  static TextStyle _createCommonTextStyle({
    required String fontFamily,
    required Color color,
    required FontStyle fontStyle,
    required FontWeight fontWeight,
    required double fontSize,
    required double height,
    required TextLeadingDistribution leadingDistribution,
  }) {
    return TextStyle(
      fontFamily: fontFamily,
      color: color,
      fontStyle: fontStyle,
      fontWeight: fontWeight,
      fontSize: fontSize,
      height: height,
      leadingDistribution: leadingDistribution,
    );
  }

Reply:
@phoenixit99  ,  Understood. This is a valid point, and I’ll address it in a separate issue at a more appropriate time, aligned with  issue [#34](https://github.com/pactus-project/pactus-gui/issues/34)